src/components/designer/LibraryRoadSignsButton.jsx

```diff
@@ -0,0 +1,153 @@
+import { useSearchParams } from "react-router-dom";
+
+const LibraryRoadSignsButton = ({
+  connectionId,
+  setCurrentMode,
+  setIsdragging,
+}) => {
+  // const signLibMode = useSelector(getTempSignsLibMode);
+  const dispatch = useDispatch();
+  const [open, setOpen] = useState(false);
+  const [urlParams, setUrlParams] = useSearchParams();
+  const handleMenuClick = (e) => {
+    if (e.key === "3") {
+      setOpen(false);
+    }
+  };
+  const handleOpenChange = (nextOpen, info) => {
+    if (info.source === "trigger" || nextOpen) {
+      setOpen(nextOpen);
+    }
+  };
+
+  const items = [
+    {
+      label: (
+        <>
+          <div
+            onClick={() => {
+              dispatch(storeSignsLibMode("detached"));
+              setCurrentMode("detached");
+              setIsdragging(true);
+              const url =
+                window.location.origin +
+                "/verkehrszeichenkataster/#/verkehrszeichenbibliothek" +
+                `?channel=${connectionId}`;
+              window.open(url, "_blank", "popup=yes");
+            }}
+          >
+            Getrennt
+          </div>
+        </>
+      ),
+      key: "1",
+    },
+    {
+      label: (
+        <>
+          <div
+            onClick={() => {
+              dispatch(storeSignsLibMode("timeline"));
+              setCurrentMode("timeline");
+              // setUrlParams({ channel: connectionId, mode: "timeline" });
+            }}
+          >
+            Zeitleiste
+          </div>
+        </>
+      ),
+      key: "2",
+    },
+    {
+      label: (
+        <>
+          <div
+            onClick={() => {
+              dispatch(storeSignsLibMode("overlay"));
+              setCurrentMode("overlay");
+              // setUrlParams({ channel: connectionId, mode: "overlay" });
+            }}
+          >
+            Overlay
+          </div>
+        </>
+      ),
+      key: "3",
+    },
+    {
+      label: (
+        <>
+          <div
+            onClick={() => {
+              dispatch(storeSignsLibMode("inside"));
+              setCurrentMode("inside");
+              // setUrlParams({ channel: connectionId, mode: "inside" });
+            }}
+          >
+            Innerhalb
+          </div>
+        </>
+      ),
+      key: "4",
+    },
+    {
+      label: (
+        <>
+          <div
+            onClick={() => {
+              dispatch(storeSignsLibMode("none"));
+              setCurrentMode("none");
+              setUrlParams({ channel: connectionId, mode: "none" });
+            }}
+          >
+            Schließen
+          </div>
+        </>
+      ),
+      key: "5",
+    },
+  ];
+  return (
+    <Dropdown
+      menu={{
+        items,
+        onClick: handleMenuClick,
+      }}
+      onOpenChange={handleOpenChange}
+      open={open}
+      style={{ flexShrink: 0, marginLeft: "auto" }}
+    >
+      <a onClick={(e) => e.preventDefault()}>
```

Comment:
What is the reason for the <a> tag here? You want to use the <a> tag as a redirection tool with the href prop and don't want to use onClick. And what does the onClick even do here?